modules/python-modules/syslogng/modules/hypr/__init__.py

```diff
@@ -0,0 +1,467 @@
+#############################################################################
+# Copyright (c) 2022 novaSOC
+# Copyright (c) 2022 Balazs Scheidler <bazsi77@gmail.com>
+# Copyright (c) 2023 Attila Szakacs
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+#
+# This was the original license on the code taken from:
+#     https://github.com/danelder/syslogng-hypr
+#
+"""
+Copyright (c) 2022 novaSOC
+
+Use of this source code is governed by an MIT-style
+license that can be found in the LICENSE file or at
+https://opensource.org/licenses/MIT.
+
+Original development by Dan Elder (delder@novacoast.com)
+Syslog-ng python fetcher for Hypr API (https://apidocs.hypr.com)
+
+Additional documentation available at:
+https://www.syslog-ng.com/technical-documents/doc/syslog-ng-open-source-edition/3.36/administration-guide/26#TOPIC-1768583
+"""
+
+import time
+import json
+import logging
+import base64
+import re
+from datetime import datetime, timezone
+import requests
+
+import syslogng
+
+
+class HyprAuditSource(syslogng.LogFetcher):
+    """
+    class for python syslog-ng log fetcher
+    """
+
+    # mandatory settings
+    url = None
+    bearer_token = None
+    rp_app_id = None
+    persist_name = None
+
+    # optional settings
+    page_size = 100
+    initial_hours = 4
+
+    # state
+    start_time = 0
+    end_time = 0
+
+    #
+    logger = logging.getLogger('hypr')
+    persist = None
```

Comment:
These are class attributes, which are then shadowed by instance attributes.
Creating a constructor would probably be cleaner.

Reply:
Yes, this is ugly, but works. Fortunately `logs` is not created on class level.